### DISCUSSION TITLE
Support for Aurora Serverless v1 and v2

### DISCUSSION THREAD

**Discussion by iantybo** [gruntwork-io/knowledge-base#520] [Help] [answered]
Do we have any support for Aurora Serverless, either v1 or v2? 

The Service Catalog module is still pinned to block the v4 AWS provider: https://github.com/gruntwork-io/terraform-aws-service-catalog/blob/master/modules/data-stores/aurora/main.tf#L18.

However, the Aurora reusable module does not pin the provider to pre-v4: https://github.com/gruntwork-io/terraform-aws-data-storage/blob/master/modules/aurora/main.tf#L13.

---

<ins datetime="2022-07-26T00:16:25Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109057">Tracked in ticket #109057</a></p>
</ins>

**Comment by yorinasub17** [comment, marked as answer]
The `aurora` module supports serverless mode with v3 of the AWS provider. You can set the `engine_mode` input variable to `"serverless"` to deploy in serverless mode.

**Comment by marko-kostic-scribe** [comment]
When the `engine_mode` is set to `serverless`, the resulting cluster is using the V1 stack. It seems that Serverless V2 isn't supported yet. Is this true?

**Reply by andi-pangeran** [reply]
same questions, Is Amazon Aurora Serverless v2 supported?..

in jakarta region, v1 is not available, and i got this error.. 

```
╷
│ Error: error creating RDS cluster: InvalidParameterValue: The engine mode serverless you requested is currently unavailable.
│ 	status code: 400, request id: 96d6dad6-7507-4b21-8159-657111e9d754
│
│   with module.database.aws_rds_cluster.cluster[0],
│   on .terraform/modules/database/modules/aurora/main.tf line 53, in resource "aws_rds_cluster" "cluster":
│   53: resource "aws_rds_cluster" "cluster" {
│
╵
ERRO[0023] 1 error occurred:
	* exit status 1
```
